sdk/translation/azure-ai-translation-text/src/main/java/com/azure/ai/translation/text/TextTranslationClientBuilder.java

```diff
@@ -342,11 +361,19 @@ private HttpPipeline createHttpPipeline() {
         policies.add(new AddDatePolicy());
         if (tokenCredential != null) {
             policies.add(new BearerTokenAuthenticationPolicy(tokenCredential, DEFAULT_SCOPE));
+            if (this.region != null && this.azureResourceId != null) {
+                HttpHeaders aadHeaders = new HttpHeaders();
+                aadHeaders.put(OCP_APIM_RESOURCE_ID_KEY, this.azureResourceId);
+                aadHeaders.put(OCP_APIM_SUBSCRIPTION_REGION, this.region);
+                policies.add(new AddHeadersPolicy(aadHeaders));
+            }
```

Comment:
Will AAD auth work if these headers are not set?

Reply:
Without those headers the AAD authentication will be failing for general MT endpoint. It will still work for custom endpoint.